src/plugins/management/public/components/feature_cards/feature_cards.tsx

```diff
@@ -30,26 +30,20 @@ export const FeatureCards = ({
   pageDescription,
   navigationUI: { HeaderControl },
 }: FeatureCardsProps) => {
-  const itemsPerRow = 4;
   const groupedCardForDisplay = useMemo(() => {
-    const grouped: Array<{ category?: AppCategory; navLinks: ChromeNavLink[][] }> = [];
+    const grouped: Array<{ category?: AppCategory; navLinks: ChromeNavLink[] }> = [];
     // The navLinks has already been sorted based on link / category's order,
     // so it is safe to group the links here.
     navLinks.forEach((link) => {
       let lastGroup = grouped.length ? grouped[grouped.length - 1] : undefined;
-      if (!lastGroup || lastGroup.category !== link.category) {
-        lastGroup = { category: link.category, navLinks: [[]] };
+      if (!lastGroup || lastGroup.category?.id !== link.category?.id) {
```

Comment:
Are we okay with both `lastGroup.category?.id` and `link.category?.id` being `undefined`? Consider a scenario where `lastGroup.category` is undefined but `link.category` is defined but `link.category.is` is not.

Reply:
Yes, we are OK with that. As long as the category.id is null, we will regard those as links without category, it complies with the behavior in left nav.